gravitee-am-reporter/gravitee-am-reporter-kafka/src/main/java/io/gravitee/am/reporter/kafka/kafka/KafkaJsonSerializer.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.reporter.kafka.kafka;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.confluent.kafka.serializers.json.KafkaJsonSchemaSerializer;
+
+public class KafkaJsonSerializer<T> extends KafkaJsonSchemaSerializer<T> {
+    public KafkaJsonSerializer() {
+        super();
+        configureObjectMapper();
+    }
+
+    private void configureObjectMapper() {
+        ObjectMapper mapper = this.objectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        mapper.disable(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS);
+        mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
```

Comment:
is there a reason we don't want these settings when we're not using schemas (i.e. in `JacksonSerializer`)

Reply:
Same was in non-registry version. I will keep it to have compatibility.

Reply:
what is the behaviour of the KafkaJsonSerializer if the registry url is not present, doesn't it simply serialize the content ? If it is the same output as the current Jackon maybe we can just keep the kafka one :thinking:

Reply:
KafkaJsonSchemaSerializer checks if schema.registry.url is present. If not, then it will throw an exception. This is the flow of KafkaJsonSchemaSerializer: connect to schema registry -> fetch id of schema or register new schema and fetch id of schema -> attach schemaId on the beggining of message -> convert object to json format and attach to message. https://github.com/confluentinc/schema-registry/blob/4e09a7c01e34b5a8509306ae7b70f9eb804ae9ca/json-schema-serializer/src/main/java/io/confluent/kafka/serializers/json/AbstractKafkaJsonSchemaSerializer.java#L161C5-L165C40

Reply:
We cannot use the same serializer, because Jackson only convert POJO into json string